File: app/src/main/java/com/example/tribecrypto/database/watchlist/WatchListDao.java
package com.example.tribecrypto.database.watchlist;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.tribecrypto.data.entity.WatchListEntity;

import java.util.List;

import io.reactivex.Observable;

@Dao
public interface WatchListDao {

    @Query("SELECT * FROM watch_list")
    Observable<List<WatchListEntity>> getAllWatchList();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAll(List<WatchListEntity> data);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertEntity(WatchListEntity data);

    @Update
    void updateWatchListCurrencies(List<WatchListEntity> data);

    @Update
    void updateWatchListCurrency(WatchListEntity data);

    @Query("DELETE FROM watch_list WHERE name LIKE :name")
    void deleteEntityByName(String name);

    @Query("DELETE FROM watch_list")
    void deleteAllWatchList();

}

File: app/src/main/java/com/example/tribecrypto/database/watchlist/WatchListDao.kt
package com.example.tribecrypto.database.watchlist

import androidx.room.*
import com.example.tribecrypto.data.entity.WatchListEntity
import io.reactivex.Observable

@Dao
interface WatchListDao {

    @Query("SELECT * FROM watch_list")
    fun getAllWatchList(): Observable<List<WatchListEntity>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(data: List<WatchListEntity>)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertEntity(data: WatchListEntity)

    @Update
    fun updateWatchListCurrencies(data: List<WatchListEntity>)

    @Update
    fun updateWatchListCurrency(data: WatchListEntity)

    @Query("DELETE FROM watch_list WHERE name LIKE :name")
    fun deleteEntityByName(name: String)

    @Query("DELETE FROM watch_list")
    fun deleteAllWatchList()
}